src/playercards/DeckBuilder.ttslua

```diff
@@ -0,0 +1,543 @@
+
+  playerCardPanelApi.spawnCardsByType({
+    cardList = cardList,
+    name = "customDeck",
+    startPos = "other",
+    spread = true,
+    spreadCols = 20,
+    targetState = "default"
+  })
+  Wait.frames(function() placingDeck = false end, 5)
+end
+
+function removeDeck()
+  deckObject = nil
+  playerCardPanelApi.deleteAll("default")
+  playerCardPanelApi.enable()
+end
+
+function exportDeck(_, _, _) 
+  if(deckObject and investigatorData) then
+    local deckList = {
+      slots = { },
+      investigatorId  = "",
+      bondedList = { },
+      customizations = { },
+      loadAltInvestigator = "normal"
+    }
+    -- Handle investigator
+    deckList.investigatorId = investigatorData.id
+    -- TODO: Handle alternate art
+    for _, card in ipairs(deckObject.getObjects()) do
+      -- Handle basic card addition
+      local cardMetadata = JSON.decode(card.gm_notes)
+      local cardId = cardMetadata.id
+      if deckList.slots[cardId] then
+        deckList.slots[cardId] = deckList.slots[cardId] + 1
+      else
+        deckList.slots[cardId] = 1
+      end
+      -- Handle bonded cards
+      if cardMetadata.bonded then
+        for _, bond in ipairs(cardMetadata.bonded) do
+          if deckList.bondedList[bond.id] then
+            deckList.bondedList[bond.id] = deckList.bondedList[bond.id] + bond.count
+          else
+            deckList.bondedList[bond.id] = bond.count
+          end
+        end
+      end
+      -- TODO: Handle customizable cards
+    end
+    deckTokenData.GMNotes = JSON.encode(deckList)
+    deckTokenData.Nickname = deckObject.getName()
+    spawnObjectData({
+      data = deckTokenData
+    })
+    destroyObject(deckObject)
+    destroyObject(investigatorObject)
+  else
+    broadcastToAll("Missing Deck and/or Investigator", Color.Red)
+  end
+end
+
+function importDeck(coinObject)
+  local deckData = JSON.decode(coinObject.getGMNotes())
+  destroyObject(coinObject)
+  local cardsToSpawn = {}
+  for cardId, cardCount in pairs(deckData.slots) do
+    local card = allCardsBagApi.getCardById({ id = cardId })
+    if card ~= nil then
+      for i = 1, cardCount do
+        table.insert(cardsToSpawn, { data = card.data, metadata = card.metadata})
+      end
+    end
+  end
+  Spawner.spawnCards(
+    cardsToSpawn,
+    { x = 53.22, y = 2, z = 46.46},
+    { x = 0.00, y = 270.00, z = 0.00},
+    true,
+    function() end
+  )
+  log(allCardsBagApi.getCardById({ id = deckData.investigatorId }))
+  Spawner.spawnCards(
+    {allCardsBagApi.getCardById({ id = deckData.investigatorId })},
+    { x = 60.92,  y = 2, z = 46.51},
+    { x = 0.00, y = 270.00, z = 0.00}
+  )
+end
+
+function placeCardsBasic(_, _, alt_click)
+  placeCards(false, alt_click)
+end
+
+function placeCardsUpgraded(_,_, alt_click)
+  placeCards(true, alt_click)
+end
+
+function placeCards(isUpgraded, alt_click)
+  if (alt_click) then
+    --Clear all cards
+    playerCardPanelApi.deleteAll("middle")
```

Comment:
Instead of having the whole functions in this big if statement, add a return to the first "path":
```
if (alt_click) then
    --Clear all cards
    playerCardPanelApi.deleteAll("middle")
    return
end
...
```